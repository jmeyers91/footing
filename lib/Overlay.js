"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var react_dom_1 = __importDefault(require("react-dom"));
var styled_components_1 = __importStar(require("styled-components"));
var Column_1 = __importDefault(require("./Column"));
var defaultColor = 'rgba(255, 255, 255, 0.7)';
var overlayRootEl = document.createElement('div');
overlayRootEl.style.position = 'absolute';
overlayRootEl.style.left = '0';
overlayRootEl.style.top = '0';
overlayRootEl.style.zIndex = '1000';
document.body.appendChild(overlayRootEl);
function Overlay(props) {
    var fixed = props.fixed, color = props.color, rest = __rest(props, ["fixed", "color"]);
    if (fixed) {
        return react_dom_1.default.createPortal(react_1.default.createElement(Column_1.default, __assign({}, rest, { "data-overlay": true })), overlayRootEl);
    }
    else {
        return react_1.default.createElement(Column_1.default, __assign({}, rest, { "data-overlay": true }));
    }
}
exports.Overlay = Overlay;
function isOverlay(target) {
    return target && target.dataset && target.dataset.overlay;
}
exports.isOverlay = isOverlay;
exports.default = styled_components_1.default(Overlay)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  position: absolute;\n  left: 0;\n  top: 0;\n  overflow-x: hidden;\n  overflow-y: auto;\n\n  ", "\n"], ["\n  position: absolute;\n  left: 0;\n  top: 0;\n  overflow-x: hidden;\n  overflow-y: auto;\n\n  ",
    "\n"])), function (props) {
    var backgroundColor = props.color || defaultColor;
    return props.fixed
        ? styled_components_1.css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n          width: 100vw;\n          height: 100vh;\n          background-color: ", ";\n        "], ["\n          width: 100vw;\n          height: 100vh;\n          background-color: ", ";\n        "])), backgroundColor) : styled_components_1.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n          width: 100%;\n          height: 100%;\n          background-color: ", ";\n        "], ["\n          width: 100%;\n          height: 100%;\n          background-color: ", ";\n        "])), backgroundColor);
});
var templateObject_1, templateObject_2, templateObject_3;
//# sourceMappingURL=Overlay.js.map