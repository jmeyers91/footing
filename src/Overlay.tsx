import React, { ComponentProps } from 'react';
import ReactDOM from 'react-dom';
import styled, { css } from 'styled-components';
import Column from './Column';

const defaultColor = 'rgba(255, 255, 255, 0.7)';

export interface Props extends ComponentProps<typeof Column> {
  /**
   * Whether or not to render the overlay directly inside the `body`.
   */
  fixed?: boolean;

  /**
   * Root element to render the overlay into. Defaults to `document.body`.
   */
  fixedRoot?: HTMLElement;

  /**
   * The background color of the overlay. Defaults to `rgba(255, 255, 255, 0.7)`.
   */
  color?: string;
}

export function Overlay(props: Props) {
  const { fixed, color, fixedRoot, ...rest } = props;

  if (fixed) {
    return ReactDOM.createPortal(
      <Column {...rest} data-overlay />,
      fixedRoot || document.body
    );
  } else {
    return <Column {...rest} data-overlay />;
  }
}

export function isOverlay(target: HTMLElement) {
  return target && target.dataset && target.dataset.overlay;
}

export default styled(Overlay)<Props>`
  position: absolute;
  left: 0;
  top: 0;
  overflow-x: hidden;
  overflow-y: auto;

  ${props => {
    const backgroundColor = props.color || defaultColor;
    return props.fixed
      ? css`
          width: 100vw;
          height: 100vh;
          background-color: ${backgroundColor};
        `
      : css`
          width: 100%;
          height: 100%;
          background-color: ${backgroundColor};
        `;
  }}
`;
