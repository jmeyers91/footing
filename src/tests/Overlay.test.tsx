import React from 'react';
import { render } from '@testing-library/react';
import Overlay from '../../src/Overlay';

describe('Overlay', () => {
  test('Should match snapshot', () => {
    const { container } = render(<Overlay />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
