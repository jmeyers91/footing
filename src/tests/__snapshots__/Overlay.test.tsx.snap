// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Should match snapshot 1`] = `
<div
  class="sc-bwzfXH flyiBm sc-bdVaJa eUEwfn"
  data-overlay="true"
/>
`;
